### PR TITLE
Test types that use native.CArray

### LINKED ISSUE #89: Fix remaining usage of ctc->getTypeSize()
While working on https://github.com/jonas/scala-native-bindgen/commits/shared-ptr I noticed at least another use where the bit length returned by `ctc->getTypeSize()` was not converted to bytes.

The goal of this ticket is to review all call sites and fix them, ideally with tests and a helper to do the conversion in a single place.

### PR DESCRIPTION
**Pull request by kornilova-l**
Closes #89

### CODE REVIEW

File: tests/samples/src/test/scala/org/scalanative/bindgen/samples/StructTests.scala [hunk 1]
```diff
@@ -10,5 +11,9 @@ object StructTests extends TestSuite {
       assert(point.x == 10)
       assert(point.y == 20)
     }
+
+    'bigStructSize - {
+      assert(Struct.getBigStructSize() == sizeof[Struct.struct_bigStruct])
```
Comment (jonas) [inline tests/samples/src/test/scala/org/scalanative/bindgen/samples/StructTests.scala]: :+1:

File: tests/samples/src/test/scala/org/scalanative/bindgen/samples/UnionTests.scala [hunk 2]
```diff
@@ -13,6 +13,8 @@ object UnionTests extends TestSuite {
         assert(!structPtr.i == 10)
         Union.setLongValue(structPtr)
         assert(!structPtr.l == 10000000000L)
```
Comment (jonas) [inline tests/samples/src/test/scala/org/scalanative/bindgen/samples/UnionTests.scala]: Just noticed that the name could be better: unionPtr

File: tests/samples/src/test/scala/org/scalanative/bindgen/samples/UnionTests.scala [hunk 3]
```diff
@@ -13,6 +13,8 @@ object UnionTests extends TestSuite {
...
+
+        assert(Union.getUnionSize() == sizeof[Union.union_values])
```
Comment (jonas) [inline tests/samples/src/test/scala/org/scalanative/bindgen/samples/UnionTests.scala]: Let's put it in a separate test case.

File: tests/samples/Struct.scala [hunk 4]
```diff
@@ -8,7 +8,9 @@ import scala.scalanative.native._
 object Struct {
   type struct_point = native.CStruct2[native.CInt, native.CInt]
   type point_s = native.Ptr[struct_point]
+  type struct_bigStruct = native.CArray[Byte, native.Nat.Digit[native.Nat._1, native.Nat.Digit[native.Nat._1, native.Nat._2]]]
```
Comment (jonas) [inline tests/samples/Struct.scala]: Interesting, so this doesn't the helper with getters and setters. We should probably create a ticket for that.

**Comment by kornilova-l** [comment]
Found this issue #91
So to check inner anonymous struct I used Scala Native api for setting fields:
```scala
// ...
!anonymousStruct._1 = 'a'
val array = anonymousStruct.cast[Ptr[CArray[Byte, Nat._8]]]
!structWithAnonymousStruct._2 = !array
```

**Pull request merged**